src/components/common/GProfilerModal.tsx

```diff
@@ -0,0 +1,465 @@
+            background: '',
+        };
+
+        var gProfilerQueryString = Object.keys(gProfilerQueryData)
+            .map((key) => {
+                return (
+                    encodeURIComponent(key) +
+                    '=' +
+                    encodeURIComponent(gProfilerQueryData[key])
+                );
+            })
+            .join('&');
+
+        return 'https://biit.cs.ut.ee/gprofiler/gost?' + gProfilerQueryString;
+    };
+
+    handleClickCreateGProfilerLink = async () => {
+        const {
+            topNumFeatures,
+            gProfilerToken,
+            selectedOrganism,
+            selectedSortBy,
+        } = this.state;
+        if (
+            (gProfilerToken === null || gProfilerToken === '') &&
+            selectedOrganism === null
+        ) {
+            this.setState({
+                error: 'Please select an organism',
+                gProfilerURL: null,
+            });
+            return;
+        }
+        if (topNumFeatures.length == 0) {
+            this.setState({
+                error:
+                    'No gene list selected. One gene list is at least required. ',
+                gProfilerURL: null,
+            });
+            return;
+        }
+        if (topNumFeatures.length == 0) {
+            this.setState({
+                error:
+                    'No gene list selected. One gene list is at least required. ',
+                gProfilerURL: null,
+            });
+            return;
+        }
+
+        const sortedFeatureMetricTable = this.featureMetricTable.sort(
+            (a, b) => b[selectedSortBy] - a[selectedSortBy]
+        );
+        const topFeatureQuery = this.buildFeatureQuery(
+            topNumFeatures,
+            sortedFeatureMetricTable
+        );
+        const organism =
+            gProfilerToken !== null ? gProfilerToken : selectedOrganism;
+        const gProfilerLink = this.buildGProfilerLink(
+            organism,
+            topFeatureQuery
+        );
+        if (gProfilerLink.length > 8000) {
+            this.setState({
+                error:
+                    'Too many genes in total. Try to select a combination of gene lists with fewer genes.',
+                gProfilerURL: null,
+            });
+            return;
+        }
+        this.setState({
+            gProfilerURL: gProfilerLink,
+        });
+    };
+
+    setAvailableOrganisms(availableOrganisms) {
+        const _availableOrganisms = availableOrganisms
+            .map((availableOrganism, idx: number) => {
+                return {
+                    key: idx + 1,
+                    text: availableOrganism['display_name'],
+                    value: availableOrganism['id'],
+                };
+            })
+            .sort((a, b) => (a['text'] < b['text'] ? -1 : 1));
+        this.setState({ availableOrganisms: _availableOrganisms });
+    }
+
+    async fetchAvailableOrganisms() {
```

Comment:
If this fails for some reason, is there a good way to present a message to the user? Also if this fails, gProfiler is likely down, so it might be a good idea to disable the button, and show a message that is it down.